mantidproject/mantid
# Close and review all leak issues with Poco::XML document, nodelist, etc. objects

This addresses [#10637](http://trac.mantidproject.org/mantid/ticket/10637) and hopefully closes all remaining Poco::XML issues. 

As part of this a few small remaining leaks have been fixed with AutoPtr or release(). I also removed a good bunch of unused poco header includes (Poco/File, Fath, NodeIterator, NodeFilter, etc. very frequently included and much less frequently used in practice) that seem to have been copy-pasted in multiple places. I also changed some unit tests to use AutoPtr. While these were not terrible issues, I think it is better to avoid bad code examples in unit tests.

**To test**:
- Code review
- Make sure all tests pass
- Overall, this should give the final touch to the Poco-XML clean-up (see "blocked by" tickets), but of course there can always be more issues
- Try to catch any remaining issue
